rust/sbp2json/src/lib.rs

```diff
@@ -99,9 +99,9 @@ where
     I: Iterator<Item = std::result::Result<M, E>> + 'a,
     E: std::error::Error + 'a,
 {
-    if fatal_errors {
-        Box::new(messages.take_while(|m| m.is_ok()).map(|m| m.unwrap()))
-    } else {
-        Box::new(messages.filter_map(|m| m.ok()))
-    }
+    let log_err = move |e| match fatal_errors {
```

Comment:
clippy will complain about this code, just use an `if` statement: https://rust-lang.github.io/rust-clippy/master/#match_bool